### ISSUE TITLE
Body Relocation + Sit = Cell Bug

### ISSUE THREAD

**Issue by kyeme** [HerculesWS/Hercules#844]
Original Topic: http://herc.ws/board/tracker/issue-8489-body-relocation-sit-cell-bug/?gopid=24396#entry24396

I am using pre-RE

There seems to be 3 bugs arising from this, and all of them seems to be affecting on how the client perceive where the players's position are.
1. The caster's final position after using body relocate + sit is not getting updated to the client of the caster, but is getting updated to the others.
- The reason why the player in the video keep asking that he is getting detected by my Ruwach even though he is NOT where I am using the Ruwach

2. The caster's final position after using body relocate + sit is not getting updated to the other players, but is getting updated to the caster.
- The reason why the caster knows where should I use Ruwach instead of where I thought I have him in my screen

3. The final position of the caster who use body relocate + site is not getting updated to the caster and to the other players.
- The reason why (at the last part) when he casted asura on me, we both said "he jumped/flew", so we both perceive that he is not actually beside me.

The video can be downloaded here:
https://www.mediafir...1tuj13diu0xrh4m

This bug seems to be well known on most private servers since a player from other private server told this to me. LOL. They are calling it "cell bug".

Video: https://youtu.be/BihyM5pOSwI

**Comment by anacondaqq** [comment]
xD i was report about this shit in 2012, or even in 2011, nobody listen to me... :DDD
interesting, how many other critical bugs you will find reported by me at past? (this is not adressed to topic starter)

**Comment by kyeme** [comment]
Up~ :D

**Comment by kyeme** [comment]
Any solution or temporary solution please? :[

**Comment by kyeme** [comment]
Here's my video:
https://youtu.be/5o6blCN4TZw

**Comment by MishimaHaruna** [comment]
What options do we have here? Reject sitting and possibly a few other requests, for a short time after body relocation?

What's the behavior on official servers by the way? They're also affected?

**Comment by kyeme** [comment]
I can reproduced on official :[ 
I will go for disable sitting

**Comment by kyeme** [comment]
~ Bump ~

**Comment by anacondaqq** [comment]
This is not official befavior. This is bug at official servers too!

**Comment by kyeme** [comment]
Up

**Comment by kyeme** [comment]
~ Bump ~

**Comment by kyeme** [comment]
~ Up ~

**Comment by ZiriusPH** [comment]
i propose

battle.reject_sitting_after_asura_use "bool"
battle.reject_sitting_after_asura_use_delay "x"

where x is number of milliseconds for /sit to be available after using asura strike.
default to 1 second.

Note that, we cannot completely reject the use of /sit after Body Relocate, because of Sling Shot from Genetics, and the possibility that a Sura used Body relocate just before getting hit by Sling shot.

**Comment by kyeme** [comment]
@Playtester :kiss:

**Comment by Playtester** [comment]
Well as said that's official behavior, I can reproduce it on official servers. Can always put a delay for sitting in your server, but I don't do custom implementations sorry.

**Comment by kyeme** [comment]
@Playtester 

I know you want as much as possible to replicate what we have in official. Even though this bug is working in Official still maybe we could make a config that would enable/disable this bug. In that case Players like us who don't want to see such bug will be happy.

**Comment by Playtester** [comment]
Yeah there are many improvements you can do, I wouldn't mind that. I just don't have time for it.

**Comment by hemagx** [comment]
yea, let's fix this shit boys.

**Comment by thanna** [comment]
bump. would love if this problem is fixed or if there is a work around on this.

**Comment by anacondaqq** [comment]
no, they don't care @thanna, don't waste your time by waiting official fix from devs, do some payment for them, or hire anybody which will fix the bug for you. Or you can fix it by yourself by putting extra delays or even blocking sitting on map.

**Comment by hemagx** [comment]
@anacondaqq 
Hercules is an free (as in freedom) open-source project, which is released under GNU GPL license and completely donation and commercial material free, also Hercules Developers contribute their work and time for free with no kind of advantage or extra services.
which means that their ability to work and give contribution to community is very depending on their own beliefs, free time, and what they would like to do.
if you believe they can't fix some issues presented, cause they're not available enough to keep up with the project feel free to contribute, if you find Hercules doesn't suite your needs, then feel free to find other options which suites your needs, whining about a free project is nothing but disrespect to yourself before disrespect to people contribute their work and time for free while expecting nothing in return.

@thanna 
while there may be an fix already discussed but this is more like an client issue more than server side issue, and honestly the current discussed solution doesn't look totally fine with me at least.
however i would let @MishimaHaruna decide if we should add option for sitting delay.
iirc, i tried before with @kyeme to try some more reliable fix for the issue but never worked out.

**Comment by anacondaqq** [comment]
@hemagx
> "... bla bla bla ...if something you don't like, shutup because we using GPL, and f*** off"

No man, sorry, i really tired to listen same bullshit. You can say about it to kids which has no brains, but don't try to cheat me. GPL and OpenSource it's business model where you taking money by developing features for other's, while administrators of private servers doing it for players and manage everything for them, you are (and not only) doing it for administrators of PR servers. I'm not against it, but i don't like when someone trying to pee in my face by saying really trash info for kids and cover their ass like "we are angels here, and we doing everything here just because we are altruists", no, this is not correct at all. There are maybe 1% of all people around Ragnarok Online development community doing something for altruism, but mostly 99% of people doing the job here because of their own motivation needs (money, education, feeling good because of new and complicated experience). Just one example of all your words about "freedom and angel paradise here": http://herc.ws/board/topic/7124-hemagxs-service/

I hope you will not answer for not making drama here, but it's your choise, i can argue each of mine word with facts and proofs, just depends on the situation or human i will need some time. For people who wish to join here and said anything against me and my point of view -> welcome, but i guess this is not good place for that, because it's issue tracker.


Why i'm emotional saying all of text above to you here? Because re-read my post above one more time, and you are not first guy who doing it, and you are not last one. I said: "developers don't care, and don't wait guy for official fix", and try to argue this string. It's not saying that developers good or bad, or something else, i just said what i said, if you can't understand it, sorry, maybe my english too bad, but most of my friends can understand even my english bad english.

You have not even tried to talk with people, because you are don't care about the problem. This bug exist in RO for very long time, i have reported (not by this nickname) several times about the bug, my collegues from different server's did the same thing several times too, a lot of bug-reporters here did it several times TOO, and all of reports -> just ignored for years. I remember when i first time report about the bug, it was 2011 or 2012 year, and now is 2017 year. So for minimum of 5 years no even answer about the problem. Is not such behavior not telling to anybody else who experience the problem just ANYTHING, like "hey, we will not fix it, sorry", or "we investigate and will fix it, but you should do a, b, c". But ofcourse no, because don't care and don't need it. Where am i wrong? Or you start to blame me and say something just because i said truth?

> if you believe they can't fix some issues presented, cause they're not available enough to keep up with the project feel free to contribute

What am i'm doing here at issues at hercules / and rathena for years, is not calling contributing? Or a lot of ignored issue reporters and bugreporters which spending their own time for reporting thing which people who have knowledge and experience can fix the problem, are these people not contributors too? All "bravery" comes to devs only? Or what? Please back to community from your sky, and try time to time to respect people and community for WHO you are doing what you are doing. Without community rAthena and hercules will be dead in 1 week after. And as you said something above about "freedom /  GLP / Opensource", please after 12 years of development, you did't even tried to make ANY documentation for your code. And the threshold of the entrance of the development is extremely high. And this is intentional by people with their business model of receiving money for supporting administrators via paid services.

**Comment by MishimaHaruna** [comment]
Well, this is just a visual glitch, and it also happens on official servers. It's probably possible to work around it, but someone needs to sit, come up with a decent implementation proposal and test it, then submit it. Whether I will is unknown (but unlikely), since there are many other things in my backlog that have higher priority than this (in general, if something involves the client, I tend not to do it, since testing something that I need the client for, is extremely annoying to set up).

Now, in response to the last comments (that really disappointed me by the way), why do you expect people to do things for you (and for free) when you want them to and how you want them to? Want this fixed quickly? Fix it and submit the fix, or pay someone to do so. Or wait for someone to fix it out of their own goodwill, but in that case you can't pressure them to do so. Documentation sucks? I know that much. I wrote some, but my lifetime won't be enough to write all the documentation that you wish we had. Does it bother you? Submit some documentation then.

On a side note, although many people are here for money, or to build a fame they can then sell on the market (who am I to blame them? FIY, people need food to live, there are many people that revolve around the Hercules community and are unemployed), that isn't true for everyone. I never took one dollar here, and I don't want anyone's stinky money. But don't mistake me, I'm not here for sheer altruism either (who would be so stupid? Investing thousands of hours on something for no reason?), I'm here because I want Hercules to work reasonably well (perhaps so that I can use it for myself for example?), and I invest my time on it so that it keeps working well, and to fix the things that don't. Although, out of all the bugs that Hercules has (and it has many, along with years and years of spaghetti code that was added without ever taking a break to do some review or planning), why do you think that YOU get decide which ones take priority?

I'm locking this issue for the time being, since it got completely out of hands. If anyone wants to add anything, feel free to open a pull request, and mention this issue ID.
